modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java

```diff
@@ -536,125 +597,68 @@ public void postOrganizationAccountsByExternalReferenceCode(
+			throw new NoSuchOrganizationException(
+				"Unable to find organization with external reference code " +
+					externalReferenceCode);
 		}
 
-		AccountContactInformation accountContactInformation =
-			account.getAccountContactInformation();
-
-		if (accountContactInformation != null) {
-			_addOrUpdateContact(
-				contactId, contextUser.getUserId(),
-				AccountEntry.class.getName(), accountEntry.getAccountEntryId(),
-				null, null, null, null, 0, 0, true, 0, 1, 1970,
-				GetterUtil.getString(accountContactInformation.getSms()),
-				GetterUtil.getString(accountContactInformation.getFacebook()),
-				GetterUtil.getString(accountContactInformation.getJabber()),
-				GetterUtil.getString(accountContactInformation.getSkype()),
-				GetterUtil.getString(accountContactInformation.getTwitter()),
-				null);
-		}
-		else {
-			_addOrUpdateContact(
-				contactId, contextUser.getUserId(),
-				AccountEntry.class.getName(), accountEntry.getAccountEntryId(),
-				null, null, null, null, 0, 0, true, 0, 1, 1970, null, null,
-				null, null, null, null);
-		}
-
-		_addAddresses(accountId, account);
-
-		return _toAccount(accountEntry);
+		postOrganizationAccountsByExternalReferenceCode(
+			organization.getOrganizationId(), externalReferenceCodes);
 	}
 
 	@Override
-	public Account putAccountByExternalReferenceCode(
-			String externalReferenceCode, Account account)
+	public Account putAccount(Long accountId, Account account)
 		throws Exception {
 
-		AccountEntry accountEntry =
-			_accountEntryService.addOrUpdateAccountEntry(
-				externalReferenceCode, contextUser.getUserId(),
+		AccountEntry accountEntry = _accountEntryService.fetchAccountEntry(
+			accountId);
+
+		if (accountEntry == null) {
+			accountEntry = _accountEntryService.addAccountEntry(
+				contextUser.getUserId(),
 				_getParentAccountId(
 					account, AccountConstants.ACCOUNT_ENTRY_ID_DEFAULT),
 				account.getName(), account.getDescription(),
-				_getDomains(account), null,
-				_getLogoBytes(
-					account,
-					_accountEntryService.
-						fetchAccountEntryByExternalReferenceCode(
-							contextUser.getCompanyId(), externalReferenceCode),
-					false),
-				null, _getType(account), _getStatus(account),
+				_getDomains(account), null, _getLogoBytes(account, null, false),
+				account.getTaxId(), _getType(account), _getStatus(account),
 				_createServiceContext(account));
+		}
+		else {
+			accountEntry = _accountEntryService.updateAccountEntry(
+				accountId,
+				_getParentAccountId(
+					account,
+					GetterUtil.getLong(
+						accountEntry.getParentAccountEntryId(),
+						AccountConstants.ACCOUNT_ENTRY_ID_DEFAULT)),
+				account.getName(), account.getDescription(),
+				_isDeleteLogo(account, null), _getDomains(account),
+				accountEntry.getEmailAddress(),
+				_getLogoBytes(account, accountEntry, false), account.getTaxId(),
+				_getStatus(account), _createServiceContext(account));
+		}
+
+		_accountEntryService.updateExternalReferenceCode(
```

Comment:
I think we need to save the result in the accountEntry variable

Reply:
@smottal I would like to argue here a bit if you don't mind. I don't think it is necessary because we don't use the new (freshly saved ERC) after this line. From my point of view it just makes the code more bloated. I purposefully did this to save some characters. If I would have used the new ERC somewhere after, sure, then I completely agree.